### PR TITLE
You can now blow open secure storage containers 

### PR DESCRIPTION
**Pull request by KazeEspada**
You can now blow open secure storage containers such as digital safes, locked briefcases, etc. using C4.

All storage items now drop their contents on deletion.

:cl: KazeEspada
C4 will now blow open locked containers.
/:cl:

### CODE REVIEW

**Comment by ChangelingRain** [comment]
>blow open
>doesn't actually blow it open

?????

**Comment by KazeEspada** [comment]
I could do that, but it would be more of a special case.

**Comment by GunHog** [comment]
I do not see much point in this if it deletes the contents...

**Comment by YotaXP** [comment]
Does it leave the nuke disk on the floor, or does it get recreated?

**Comment by KazeEspada** [comment]
Nuke disc gets recreated.

**Comment by MMMiracles** [comment]
Whats the purpose of blowing it open if it'll just delete whatever inside it in the end, exactly?

**Comment by ike709** [comment]
IF I CAN'T HAVE IT, NOBODY CAN HAVE IT

**Comment by Zilenan91** [comment]
Yeah why does it destroy everything inside? What's the point in doing this if you get nothing.

**Comment by WJohn** [comment]
This doesn't make any sense, it really should give you the contents.

Also changelog, this is a neat thing to know about.

**Comment by Dorsisdwarf** [comment]
The point is to extract a nuke disk, I guess

**Comment by Jalleo** [comment]
Actually there is a good reason lemme just get a pic

File: code/game/objects/items/weapons/explosives.dm [hunk 1]
```diff
@@ -69,10 +69,16 @@
 /obj/item/weapon/c4/afterattack(atom/movable/AM, mob/user, flag)
 	if (!flag)
 		return
-	if (ismob(AM) || istype(AM, /obj/item/weapon/storage/))
+	if (ismob(AM))
 		return
 	if(loc == AM)
 		return
+	if((istype(AM, /obj/item/weapon/storage/)) && !((istype(AM, /obj/item/weapon/storage/secure)) || (istype(AM, /obj/item/weapon/storage/lockbox)))) //If its storage but not secure storage OR a lockbox, then place it inside.
+		return
```
Comment (KorPhaeron) [inline code/game/objects/items/weapons/explosives.dm]: This whole bit is not needed, you can already put c4 inside bags

File: code/game/objects/items/weapons/storage/storage.dm [hunk 2]
```diff
@@ -470,8 +470,8 @@
 
 
 /obj/item/weapon/storage/Destroy()
-	for(var/obj/O in contents)
-		O.mouse_opacity = initial(O.mouse_opacity)
+	var/turf = get_turf(src)
+	empty_object_contents(0, turf)
```
Comment (KorPhaeron) [inline code/game/objects/items/weapons/storage/storage.dm]: Why did you remove the mouse opacity thing?

File: code/game/objects/items/weapons/explosives.dm (hunk 1 above)
Comment (KazeEspada) [inline code/game/objects/items/weapons/explosives.dm]: Actually without this part, it places it on the bag.

File: code/game/objects/items/weapons/storage/storage.dm (hunk 2 above)
Comment (KazeEspada) [inline code/game/objects/items/weapons/storage/storage.dm]: Empty_object_contents handles the entire loop.
